### PR TITLE
Provide the ability to specify a default scheduler for RxJava.

### PR DESCRIPTION
**Pull request by JakeWharton**
This mimics the behavior of Retrofit 1, and can be surprisingly convenient for consistency.

### CODE REVIEW

File: retrofit-adapters/rxjava/src/main/java/retrofit2/adapter/rxjava/RxJavaCallAdapterFactory.java [hunk 1]
```diff
@@ -37,10 +38,21 @@
...
   public static RxJavaCallAdapterFactory create() {
-    return new RxJavaCallAdapterFactory();
+    return new RxJavaCallAdapterFactory(null);
   }
 
-  private RxJavaCallAdapterFactory() {
+  /**
+   * TODO
+   */
+  public static RxJavaCallAdapterFactory createWithScheduler(Scheduler scheduler) {
```
Comment (swankjesse) [inline retrofit-adapters/rxjava/src/main/java/retrofit2/adapter/rxjava/RxJavaCallAdapterFactory.java]: Do we want a builder ?

File: retrofit-adapters/rxjava/src/test/java/retrofit2/adapter/rxjava/RxJavaCallAdapterFactoryTest.java [hunk 2]
```diff
@@ -44,6 +44,15 @@
         .build();
   }
 
+  @Test public void nullSchedulerThrows() {
+    try {
+      RxJavaCallAdapterFactory.createWithScheduler(null);
+      fail();
```
Comment (swankjesse) [inline retrofit-adapters/rxjava/src/test/java/retrofit2/adapter/rxjava/RxJavaCallAdapterFactoryTest.java]: nice

**Comment by swankjesse** [comment]
Awesome.

File: retrofit-adapters/rxjava/src/main/java/retrofit2/adapter/rxjava/RxJavaCallAdapterFactory.java (hunk 1 above)
Comment (JakeWharton) [inline retrofit-adapters/rxjava/src/main/java/retrofit2/adapter/rxjava/RxJavaCallAdapterFactory.java]: I'm fighting hard against the need for one in either the converter factories or call adapter factories.

File: retrofit-adapters/rxjava/src/test/java/retrofit2/adapter/rxjava/RxJavaCallAdapterFactoryTest.java (hunk 2 above)
Comment (JakeWharton) [inline retrofit-adapters/rxjava/src/test/java/retrofit2/adapter/rxjava/RxJavaCallAdapterFactoryTest.java]: :partyhard:

**Pull request merged**